### PR TITLE
Overhauled mod/item - fixing wrong thread parent and mail stuff

### PR DESCRIPTION
**Pull request by annando**
(Hopefully) fixed a bug where it was possible that the wrong user id was selected when posting on global items.

Additionally we sometimes fetched the wrong parent. Now the items are clearly marked if they are the parent or thread parent.

And: The sending of content to your mail contacts via API hadn't worked due to a wrongly named variable.

And last but not least: The whole code of the "item_post" function was overhauled.

### CODE REVIEW

File: mod/item.php [hunk 1]
```diff
@@ -373,28 +310,19 @@ function item_post(App $a) {
...
 	if ($profile_uid == local_user() || $allow_comment) {
 		$contact_record = $author;
 	} else {
-		$r = q("SELECT * FROM `contact` WHERE `uid` = %d AND `self` LIMIT 1",
-			intval($profile_uid)
-		);
-		if (DBM::is_result($r)) {
-			$contact_record = $r[0];
-		}
+		$contact_record = dba::selectFirst('contact', [], ['uid' => $profile_uid, 'self' => true]);
```
Comment (MrPetovan) [inline mod/item.php]: Isn't `DBM::is_result()` warranted here?

File: mod/item.php [hunk 2]
```diff
@@ -579,21 +483,10 @@ function item_post(App $a) {
...
-						dbesc($str_group_allow),
-						dbesc($str_contact_deny),
-						dbesc($str_group_deny),
-						intval($profile_uid),
-						intval($attach)
-					);
-				}
+				$fields = ['allow_cid' => $str_contact_allow, 'allow_gid' => $str_group_allow,
+						'deny_cid' => $str_contact_deny, 'deny_gid' => $str_group_deny];
+				$condition = ['id' => $attach];
```
Comment (MrPetovan) [inline mod/item.php]: Why did you drop the `uid` condition?

File: mod/item.php [hunk 3]
```diff
@@ -636,15 +529,15 @@ function item_post(App $a) {
 
 	if (preg_match_all('/(\[attachment\]([0-9]+)\[\/attachment\])/',$body,$match)) {
 		foreach ($match[2] as $mtch) {
-			$r = q("SELECT `id`,`filename`,`filesize`,`filetype` FROM `attach` WHERE `uid` = %d AND `id` = %d LIMIT 1",
-				intval($profile_uid),
-				intval($mtch)
-			);
-			if (DBM::is_result($r)) {
+			$fields = ['id', 'filename', 'filesize', 'filetype'];
+			$attachment = dba::selectFirst('attach', $fields, ['id' => $mtch]);
```
Comment (MrPetovan) [inline mod/item.php]: Why did you drop the `uid` condition?

File: mod/item.php [hunk 4]
```diff
@@ -936,7 +832,7 @@ function item_post(App $a) {
 	item_post_return(System::baseUrl(), $api_source, $return_path);
 	// NOTREACHED
 }
-
+// q(
```
Comment (MrPetovan) [inline mod/item.php]: Can you please remove this comment?

File: mod/item.php (hunk 1 above)
Comment (annando) [reply mod/item.php]: In the previous coding ```$contact_record``` was undefined when the query had gone wrong. So here the variable is at least defined. It's not ideal but better than before.

File: mod/item.php (hunk 2 above)
Comment (annando) [reply mod/item.php]: The primary key is already the "id" field. Querying an additional field doesn't make any sense.

File: mod/item.php (hunk 3 above)
Comment (annando) [reply mod/item.php]: Same like before. The primary field is "id".

File: mod/item.php (hunk 4 above)
Comment (annando) [reply mod/item.php]: Oops, that was a test comment.

File: mod/item.php (hunk 1 above)
Comment (MrPetovan) [reply mod/item.php]: Gotcha.

File: mod/item.php (hunk 2 above)
Comment (MrPetovan) [reply mod/item.php]: It does to make sure the attachment is owned by `$profile_uid`, can you ensure it is?

File: mod/item.php (hunk 3 above)
Comment (MrPetovan) [reply mod/item.php]: Same like before, it made sure the attachment is owned by `$profile_uid`, can you still ensure it is? 😛

Comment (annando) [reply mod/item.php]: I had the same idea, so I changed it.

Comment (MrPetovan) [reply mod/item.php]: I've seen you added a check for the other attach condition, are you going to change this one as well?

Comment (annando) [reply mod/item.php]: You could always want to link an attachment that has already been attached before (and it could be public)

Comment (MrPetovan) [reply mod/item.php]: Ok, I'm just making sure we aren't losing a critical privacy check in the process. Are you done with your changes now?

Comment (annando) [reply mod/item.php]: Yeah, I am. But currently my system blanks - but also on the "develop" branch, you it isn't a problem of my changes.

Comment (annando) [reply mod/item.php]: Problem solved. It was in one of the addons.

**Pull request merged**